### DISCUSSION TITLE
Blog and server error 403 on Windows

### DISCUSSION THREAD

**Discussion by BsNoSi** [datenstrom/community#91] [Report a bug] [answered]
If blog is hidden (blog instead of x-blog) the folder is not part of search-plugin.

Currently I solved this with x= last no of folders and title=. (a single dot). Works for me, but this should not be the standard solution :)

Possibly this is part of the challenge, that a "1-blog" will not find tags, because "domain.tld/tag:xxx" will lead to Error 403 (@Mark: I send you this by mail).

**Comment by markseuffert** [comment]
Thanks for testing Yellow. 

Yes that's an expected behavior, only visible pages are included in the [search](https://github.com/annaesvensson/yellow-search). At least that was the idea behind it, that you can have invisible things that do not show up in navigation, search and so on. Can I ask why you want to make a hidden blog?

**Comment by BsNoSi** [comment]
Ah. See. Misleading information. 

**Scenario:**  

I want to use "blog" in first place. So I renamed "blog" to "1-blog", as described in plugin notes. This generally works fine, but after opening a blog entry, the usage of tags is prohibited: Error 403.

Reason for it is wrong link generation. Instead of "domain/blog/tag:xxx" →  "domain/tag:xxx" is created. Same with overviews instead of "domain/blog/page:xxx" > →  "domain/page:xxx" is given. This does not work.

So I was looking for a workaround. If I use "1-start" with links to blog pages (thanks for content-blogpreview sample!), and  "4-blog" with title "." (single Dot, last entry), it _looks like_ a clean blog. As a major result "tag:xxxx" or "page:xxx" works as expected. 

It does as well, if I rename "1-blog" into "blog", but then search does not operate as expected. Which is correct, because blog is hidden now (useful option). 

**Desired Scenario:**
- Using "1-blog" as entry page.
- Generating pages from the blog without "/blog/" part in url
- Tags or overview links generate tag/page overview instead of error.
- Search works for blog pages as well as for static ones.

I have to admit, that my php experience is to low, to have a usable idea for solving this. But I hope, this description has enough details for debugging.

**Comment by markseuffert** [comment]
Ah, now I understand what you want to do. Thanks for the detailed description. Let's solve the 403 error first... and then I'll help you with the blog configuration.

**Comment by BsNoSi** [comment]
Hello Mark,

I am quite sure, that my challenge is solved, as soon as you
have an idea regarding to 403.

I for myself have solved it currently with this "dot-option" (see
http://blog.nosi.de ) Its a portation from HTMLy, which was quite easy
to do.

 This an next weeks I have only sporadically time for testing and
 developing things.  So, please, be patient, if I do not reply
directly.

**Comment by markseuffert** [comment]
I am not able to reproduce the 403 error, can you give me a link?

**Comment by BsNoSi** [comment]
I´ll create a testing area for you. Link will go to your personal adress.

**Comment by markseuffert** [comment]
Looks like the error doesn't happen on your web server.

**Comment by BsNoSi** [comment]
**Some more informations.**

The „Error 403“ occurs _only_ in my local testing area driven by XAMPP. Currently it seems to me, that _all_ my noticed side effects are caused by XAMPP. As Mark mentioned, on my external test drive everything works fine. No idea for that. If I find a solution for that, I will notify it here.

**Comment by markseuffert** [comment]
Here's what I found out. 

The 403 error happens because of the Windows filesystem. It's an [Apache bug on Windows](https://bz.apache.org/bugzilla/show_bug.cgi?id=41441), which does not support if the first path segment contains a colon character. Because the colon character is used by the Windows filesystem, you can not access some links. You'll see in the Apache error log something like `(20024)The given path is misformatted or contained invalid characters`.

How to fix this? You can move your Yellow installation into a subfolder, e.g. `http://localhost/yellow/`. Please let me know if this works for you. Thanks.

**Comment by BsNoSi** [comment]
Moving into a subfolder did not help. Still same error message. I did not check, but I think, this will be generally no solution, because (afaik) a colon is generally not allowed in path description of Windows.

Since the problem is revealed, I can deal with it.  Inside of my development area these links do not need to work.

**Comment by markseuffert** [comment, marked as answer]
Should be working now, Windows support was added to latest Yellow.
